### ISSUE TITLE
Note---Intro of Machine/Deep Learning 1 by hylee

### ISSUE THREAD

**Issue by raychenv** [raychenv/blog#361]
https://www.youtube.com/watch?v=Ye018rCVvOo&list=PLJV_el3uVTsMhtt7_Y6sgTHGHp1Vb2P2J&index=1

Machine Learning ~~~ Looking for Function
- Speech Recognition
f(voice)  = "How are you"
- Image recognition
f(image) = "cat"
- Playing go
f(cheese board) = (next move)

Different types of Functions
- Regression: The function outputs a scalar
Predict PM2.5 PM2.5 today + temperature + O3 --> f --> PM2.5 of tomorrow
- Classification: Given options(classes), the function outputs the correct one
Spam filtering Email --> f --> Yes/NO
Playing go --> f --> Each position is a class Next Move
以上只是机器学习里的一小部分呀
- Structured Learning create something with structure (image, document)

A case study
1. Function with Unknown Parameters
**Model** y = b + w xi based on domain knowledge
**feature** no of views xi
**weight** w
**bais** b
2. Define Loss from training Data
**Loss** is a function of parameters L(b, w)
how good a set of value is
L(0.5k, 1) y = b + w x1 --> y = 0.5k + 1 x1 How good it is?
Loss: L = 1/N E en
e = Iy-y'I   L is mean absolute error MAE
e = (y-y')2 L is mean squre error MSE
Diagram of L??
3. Optimization
w*, b* = arg min L(w, b)
Gradient descent 梯度下降
- Pick an initial value of w
- Compute 偏微分
- Next step w1 = w0 + u 偏微分   u learning rate
- Update w iteratively to
**Local minima**
**global minima**

Machine Learning is so simple ......
step 1: function is unknown
step2: define loss from training data
step3: optimization

2017-2020 training data
2021 unseen during training

Domain knowledge
